controller/scan/registry.go

```diff
@@ -1544,29 +1550,21 @@ func (rs *Registry) scheduleScanImages(
 					scanTypesRequired.Vulnerability = true
 				}
 
-				signatureInfoChanged := info.SignatureDigest != sum.SignatureDigest
-				sigstoreConfigurationChanged := sigstoreTimestamp != sum.SigstoreTimestamp
-
-				if signatureInfoChanged {
+				// determine if signature scan is required
+				if !newImage {
+					scanTypesRequired.Signature = false
+				} else if signatureInfoChanged {
+					smd.scanLog.WithFields(log.Fields{"registry": rs.config.Name, "image": image, "status": sum.Status, "imageID": info.ID}).Debug("Signature info for image changed, signature scan required.")
+					scanTypesRequired.Signature = true
 					sum.SignatureDigest = info.SignatureDigest
 					imageChanged = true
+				} else if sigstoreConfigurationChanged && isSignedImage {
+					smd.scanLog.WithFields(log.Fields{"registry": rs.config.Name, "image": image, "status": sum.Status, "imageID": info.ID}).Debug("Sigstore config changed and image is signed, signature scan required.")
 					scanTypesRequired.Signature = true
-					smd.scanLog.WithFields(log.Fields{"registry": rs.config.Name, "image": image, "status": sum.Status, "imageID": info.ID}).Debug("Signature Info Changed")
-				}
-
-				if sigstoreConfigurationChanged {
 					sum.SigstoreTimestamp = sigstoreTimestamp
 					imageChanged = true
-					if info.SignatureDigest != "" {
-						// image is signed, rescan with new sigstore configuration
-						smd.scanLog.WithFields(log.Fields{"registry": rs.config.Name, "image": image, "status": sum.Status, "imageID": info.ID}).Debug("Sigstore Config Changed")
-						scanTypesRequired.Signature = true
-					}
-				}
-
-				if sum.SignatureStatus == api.ScanStatusFailed {
-					scanTypesRequired.Signature = true
-				} else if newImage && info.SignatureDigest != "" {
+				} else if sum.SignatureStatus == api.ScanStatusFailed {
```

Comment:
should it be
else if sum.SignatureStatus == api.ScanStatusFailed && isSignedImage {
?

Reply:
With the logic now, `sum.SignatureStatus` should only ever show a fail for a signed image anyways, since we are only scanning signed images.